### PR TITLE
Fix a bug with invalidating blocks in new DB and add more sanity checks

### PR DESCRIPTION
**Pull request by cjepson**
A bug prevented STXO data from being written correctly in the event
that a block became invalidated. This could cause failures on testnet
when a block was invalidated. There was also some debug code and a panic
that were in master that have now been removed. A new test has been
added to ensure that STXO data is being properly written, and more
stringent checks for STXO data validity when added or removing blocks
has been added.

### CODE REVIEW

File: blockchain/chainio.go [hunk 1]
```diff
@@ -479,8 +482,13 @@ func serializeSpendJournalEntry(stxos []spentTxOut) ([]byte, error) {
 	var size int
 	var sizes []int
 	for i := range stxos {
-		// size += spentTxOutSerializeSize(&stxos[i])
 		sz := spentTxOutSerializeSize(&stxos[i])
+		if sz < minimumStxoSize {
```
Comment (dajohi) [inline blockchain/chainio.go]: I think spentTxOutSerializeSize should check this and return error so caller doesn't forget.

File: blockchain/chainio.go [hunk 2]
```diff
@@ -30,6 +30,10 @@ const (
 	// currentDatabaseVersion indicates what the current database
 	// version is.
 	currentDatabaseVersion = 1
+
+	// minimumStxoSize is the absolute smallest size a serialized STXO
+	// may be, below which is assumed to be an error.
+	minimumStxoSize = 4
```
Comment (alexlyp) [inline blockchain/chainio.go]: I assume minimumStxoSize is due to 1 coinbase + 3 stakebases, correct?  Maybe add that in the comment or why it is 4.

File: blockchain/chainio.go (hunk 1 above)
Comment (cjepson) [inline blockchain/chainio.go]: It doesn't look like you can ever trigger this error, so I ended up just removing it.

**Comment by jcvernaleo** [comment]
Seems to be running happily on my two testnet nodes.

**Comment by alexlyp** [comment]
Been running well on my tip nodes.  tACK

**Comment by marcopeereboom** [comment]
That looks right.  OK

**Pull request merged**